### PR TITLE
Svg infrastructure

### PR DESCRIPTION
**Pull request by brucemiller**
Generalized the SVG support for various primitives and foreignObject, moved from the pgf driver to TeX.pool.  This allows it's use for xy binding.

### CODE REVIEW

**Comment by dginev** [comment]
I quickly pulled the branch locally and tried it on the Tikz examples from [the showcase](https://latexml.mathweb.org/editor). All looks good!

Will read through the code quickly now.

File: lib/LaTeXML/Package/TeX.pool.ltxml [hunk 1]
```diff
@@ -3484,11 +3543,27 @@ sub DimensionToSpaces {
...
-DefPrimitive('\hskip Glue', sub {
+# \hskip handled similarly to \kern
+# \hskip can be ignored in certain situations...
+DefConstructor('\hskip Glue', sub {
+    my ($document, $length, %props) = @_;
+    my $parent = $document->getNode;
+    #    Debug("HSKIP ".ToString($length)." at ".$document->getNodeQName($parent));
+    if ($document->getNodeQName($parent) eq 'svg:g') {
+      my $x = $length->pxValue;
+      # HACK HACK HACK
```
Comment (dginev) [inline lib/LaTeXML/Package/TeX.pool.ltxml:3554]: Well marked.

**Review by dginev — APPROVED** [review]
Somewhat scary, but I like it!

File: lib/LaTeXML/Package/TeX.pool.ltxml [hunk 2]
```diff
@@ -2358,7 +2401,7 @@ sub writableTokens {
 
 DefPrimitive('\message{}', sub {
     my ($stomach, $stuff) = @_;
-    Note(writableTokens(Expand($stuff)));
+    NoteLog(writableTokens(Expand($stuff)));
```
[the remark this answers is not in the document]
Comment (brucemiller) [reply lib/LaTeXML/Package/TeX.pool.ltxml:2404]: yep, and it seems that tex only puts them in the log, so...

**Pull request merged**